main/src/library/StringBuilder.flix

```diff
@@ -113,15 +117,36 @@ namespace StringBuilder {
     ///
     /// Append the system line separator to the StringBuilder `sb`.
     ///
-    pub def appendLineSeparator(sb: StringBuilder): Unit & Impure =
-        sb.appendString(String.lineSeparator())
+    pub def appendLineSeparator!(sb: StringBuilder): Unit & Impure =
+        sb.appendString!(String.lineSeparator())
 
     ///
     /// Append the String `s` followed by the system line separator to the StringBuilder `sb`.
     ///
-    pub def appendLine(sb: StringBuilder, s: Str): Unit & Impure =
-        sb.appendString(s);
-        sb.appendLineSeparator()
+    pub def appendLine!(sb: StringBuilder, s: String): Unit & Impure =
+        sb.appendString!(s);
+        sb.appendLineSeparator!()
+
+    ///
+    /// Use the function `f` to converted the element `x` to a string with `f` append it, followed by the
+    /// system line separator, to the StringBuilder `sb`.
+    ///
+    pub def appendLineWith!(sb: StringBuilder, f: a -> String, x: a): Unit & Impure =
```

Comment:
How about:

Appends `f(x)` to the string builder `sb`.

or something like that?